src/redux.js

```diff
@@ -1259,20 +947,77 @@ export function getIncorrectResults(state) {
   return getResultsByGrade(state, ResultsGrades.INCORRECT);
 }
 
-function getResultsByGrade(state, grade) {
-  const results = {};
-  const accuracyGrades = getAccuracyGrades(state);
-  const examples = getConvertedAccuracyCheckExamples(state).filter((example, index) => {
-    return grade === accuracyGrades[index];
-  });
-  const labels = getConvertedLabels(state, state.accuracyCheckLabels).filter((example, index) => {
-    return grade === accuracyGrades[index];
-  });
-  const predictedLabels = getConvertedLabels(state, state.accuracyCheckPredictedLabels).filter((example, index) => {
-    return grade === accuracyGrades[index];
-  });
-  results.examples = examples;
-  results.labels = labels;
-  results.predictedLabels = predictedLabels;
-  return results;
+/* Functions for processing data about a trained model to save. */
+
+export function isUserUploadedDataset(state) {
+  // The csvfile for internally curated datasets are strings; those uploaded by
+  // users are objects. Use data type as a proxy to know which case we're in.
+  return typeof state.csvfile === "object" && state.csvfile !== null;
+}
+
+export function getDataDescription(state) {
+  // If this a dataset from the internal collection that already has a description, use that.
+  if (
+    state.metadata &&
+    state.metadata.card &&
+    state.metadata.card.description
+  ) {
+    return state.metadata.card.description;
+  } else if (
+    state.trainedModelDetails &&
+    state.trainedModelDetails.datasetDescription
+  ) {
+    return state.trainedModelDetails.datasetDescription;
+  } else {
+    return undefined;
+  }
+}
+
+export function getDatasetDetails(state) {
+  const datasetDetails = {};
+  datasetDetails.name = state.metadata.name;
+  datasetDetails.description = getDataDescription(state);
+  datasetDetails.numRows = state.data.length;
+  datasetDetails.isUserUploaded = isUserUploadedDataset(state);
+  return datasetDetails;
+}
+
+export function getFeaturesToSave(state) {
+  const features = state.selectedFeatures.map(feature =>
+    getColumnDataToSave(state, feature)
+  );
+  return features;
+}
+
+export function getLabelToSave(state) {
+  return getColumnDataToSave(state, state.labelColumn);
+}
+
+function getSummaryStat(state) {
+  let summaryStat = {};
+  summaryStat.type = isRegression(state)
+    ? MLTypes.REGRESSION : MLTypes.CLASSIFICATION;
+  summaryStat.stat = getPercentCorrect(state);
+  return summaryStat;
+}
```

Comment:
Do we still need `getSummaryStat` if we're using `getPercentCorrect`?

Reply:
yes, because we still use in the `getDataToSave` function because we want to save the type of machine learning (regression or classification) alongside the percentage.